compiler/rustc_typeck/src/check/check.rs

```diff
@@ -382,10 +382,15 @@ fn check_union_fields(tcx: TyCtxt<'_>, span: Span, item_def_id: LocalDefId) -> b
                     tcx.sess,
                     field_span,
                     E0740,
-                    "unions may not contain fields that need dropping"
+                    "unions cannot contain fields that may need dropping"
+                )
+                .note(
+                    "a type is guaranteed not to need dropping \
+                    when it implements `Copy`, or when it is the special `ManuallyDrop<_>` type.",
```

Comment:
```suggestion
                    when it implements `Copy`, or when it is the special `ManuallyDrop<_>` type",
```
mega-nit: I don't think we normally end diagnostics with a period.

Reply:
👍 Done in https://github.com/rust-lang/rust/pull/94031/commits/43e6df1a0a4e53d5879203a1447cdb7a0657dd51 (now squashed)